README.md

```diff
@@ -112,41 +112,118 @@ Options can be specified on the command line or in `config/release.js` unless ma
 
 ## Hooks
 
-A set of lifecycle hooks exists as a means to inject additional behavior into the release process. Lifecycle hooks can
-be specified in `config/release.js`. The following lifecycle hooks are available:
+A set of lifecycle hooks exists as a means to inject additional behavior into the release process. Lifecycle hooks can be specified in `config/release.js`. All hooks can return a thenable that will be resolved before continuing the release process. Rejecting a promise returned in a hook will halt the release process and exit with an error.
+
+Hooks are passed two arguments:
+
+  - `project` - a reference to the current ember-cli project
+  - `versions` - an object containing tag information, which will always have a `next` property and depending on the strategy you are using, may also have a `latest` property.
+
+There are currently two lifecycle hooks available:
 
 - `beforeCommit`
 
-  A function to run when the new version has been computed but before the version changes have been committed. This is the hook to use if you need to update the version number in additional files besides the standard package.json and bower.json. The function is passed `project` and `versions`. `versions` will always have a `next` property and depending on the strategy you are using, may also have a `latest`
+  Called after the new version has been computed and replaced in manifest files but before the changes have been committed. Use this hook if you need to update the version number in additional files, or build the project to update dist files. Note that this hook runs regardless of whether a commit will be made.
 
   ###### Example Usage
 
+  Version replacement:
+
   ```js
   // config/release.js
   var path = require('path');
   var xmlpoke = require('xmlpoke');
 
   module.exports = {
-    beforeCommit: function(project, versions){
+    beforeCommit: function(project, versions) {
       xmlpoke(path.join(project.root, 'cordova/config.xml'), function(xml) {
         xml.errorOnNoMatches();
         xml.addNamespace('w', 'http://www.w3.org/ns/widgets');
         xml.set('w:widget/@version', versions.next);
       });
     }
-  }
+  };
   ```
 
-##### Planned Hooks
+  Building:
+
+  ```js
+  // config/release.js
+  var BuildTask = require('ember-cli/lib/tasks/build');
+
+  // For details on each option run `ember help release`
+  module.exports = {
+    // Build the project in the production environment, outputting to dist/
+    beforeCommit: function(project) {
+      var task = new BuildTask({
+        project: project,
+        ui: project.ui,
+        analytics: project.cli.analytics
+      });
```

Comment:
Yeah, agreed (forgot about the context being wrong in the hooks). 

Also, I think at some point in the future, building might be common enough for a true/false flag in the command line or config file.